src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +121,51 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(this._pollingTimer);
+        this._deinitializePollingTimer(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string) {
+        this._initializePollingTimer();
+        this._pollingInterval = window.setTimeout(async () =>  {
+            try {
+                if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status.toLowerCase() === ORDER_STATUS_APPROVED) {
+                        this._deinitializePollingTimer(this._pollingTimer);
+                        this._deinitializePollingTimer(this._pollingInterval);
+                        this._tokenizePayment({orderID: this._paypalCommercePaymentProcessor.getOrderId()}, submitForm);
+                    } else {
+                        this._initializePollingMechanism(submitForm, gatewayId);
+                    }
+                }
+            } catch (e) {
+                this._deinitializePollingTimer(this._pollingInterval);
+                this._deinitializePollingTimer(this._pollingTimer);
+            }
+        }, POLLING_INTERVAL);
+    }
+
+    private _deinitializePollingTimer(timer?: number) {
+        if (timer) {
+            clearTimeout(timer);
+            timer = 0;
+        }
+    }
+
+    private _initializePollingTimer(): void {
+        if (!this._pollingTimer) {
+            this._pollingTimer = window.setTimeout(() => {
+                this._deinitializePollingTimer(this._pollingInterval);
+                this._deinitializePollingTimer(this._pollingTimer);
+            }, POLLING_MAX_TIME);
```

Comment:
Is there a reason why you need another `setTimeout`? Is it because you want to stop polling after a minute? Wouldn't the polling be automatically cancelled if the order status becomes approved (or not)? What would happen if the polling is stopped before the approval? Would the user keep seeing the loading spinner?